File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/access/annotationprovider/CallbackAnnotationProvider.java
package ru.tggc.telegrambotcore.access.annotationprovider;

import org.springframework.stereotype.Component;
import ru.tggc.telegrambotcore.annotation.handle.CallbackHandle;
import ru.tggc.telegrambotcore.annotation.handle.HandleMeta;

import java.lang.reflect.Method;

@Component
public class CallbackAnnotationProvider implements AnnotationProvider {

    @Override
    public boolean supports(Method m) {
        return m.isAnnotationPresent(CallbackHandle.class);
    }

    @Override
    public HandleMeta extractMeta(Method m) {
        return HandleMeta.from(m.getAnnotation(CallbackHandle.class));
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/access/annotationprovider/CallbackAnnotationProvider.kt
package ru.tggc.telegrambotcore.access.annotationprovider

import org.springframework.stereotype.Component
import ru.tggc.telegrambotcore.annotation.handle.CallbackHandle
import ru.tggc.telegrambotcore.annotation.handle.HandleMeta
import java.lang.reflect.Method

@Component
class CallbackAnnotationProvider : AnnotationProvider {
    override fun supports(m: Method): Boolean = m.isAnnotationPresent(CallbackHandle::class.java)


    override fun extractMeta(m: Method): HandleMeta? =
        HandleMeta.from(m.getAnnotation(CallbackHandle::class.java))

}
